### PR TITLE
update check-mysql-replication-status, check-mysql-alive to use mysql2 gem

### PR DESCRIPTION
**Pull request by notnmeyer**
## Pull Request Checklist

**Is this in reference to an existing issue?**

nope

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass 

#### Purpose

The latest Sensu ships with Ruby 2.4. The `mysql` gem is abandoned and will not build on the embedded Ruby. This PR updates check-mysql-replication-status, and check-mysql-alive to use `mysql2 ` instead.

#### Known Compatablity Issues

none

### CODE REVIEW

**Comment by notnmeyer** [comment]
rubocop complains about a few things, none are related to this pr.

**Comment by notnmeyer** [comment]
huh, just came across some prior art in https://github.com/sensu-plugins/sensu-plugins-mysql/pull/19

admittedly, we dont install the plugins via the gem, we just manage them ourselves (i know, bad form). so hadnt noticed the inclusion of `ruby-mysql` until now. i'm not necessarily looking to reopen the `ruby-mysql` vs. `mysql2` conversation (but i would point out that ruby-mysql has not been updated for like a year and half).

mysql2 obviously does require a compiler and mysql dev packages to build, so i get how the earlier decision was arrived at.

feel free to close this if mysql2 isn't something you're interested in!

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+### Added
```
Comment (majormoses) [inline CHANGELOG.md]: As this needs a c compiler now this is a breaking change, can you please change the header to `### Breaking Changes`. Also we should include an explicit entry for this and add some information on such in the README.md

**Comment by majormoses** [comment]
@eheydrick what are your thoughts? I am hesitant to accept this for the reasons outlined in review and how many issues I have seen where people do not want to requirement of a c compiler. Minimally it needs more documentation. One approach (that I do not like) we could take is to not make it runtime dependency of them and have flags to choose deps? this is hacky but I can't think of a solution that will make everyone happy.

**Comment by eheydrick** [comment]
I'd prefer to use the `ruby-mysql` gem as 1) we've already started down that path and 2) it doesn't require compilers to install.

**Comment by majormoses** [comment]
@eheydrick the one thing that makes me want to accept this is that `ruby-mysql` is the lack of active development on it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,7 +8,17 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
...
+### Breaking Change
+- check-mysql-status.rb: renamed short arg of `--check` due to conflicts (@majormoses)
+
+### Changed
+- check-mysql-status.rb: made the options easier to read by splitting them across multiple lines (@majormoses)
+
+### Added
+- Add testing on Ruby 2.4 (@eheydrick)
+
+>>>>>>> master
```
Comment (majormoses) [inline CHANGELOG.md]: looks like leftover merge conflict.

Comment (notnmeyer) [inline CHANGELOG.md]: thanks!

**Comment by majormoses** [comment]
@sensu-plugins/commit-bit please vote on whether we should accept this as I am torn and need others thoughts. If there is no active development on `mysql-ruby` it is not a path that we can go down for long on the other hand the proposed changes require a c compiler and many people do not want that.

**Comment by notnmeyer** [comment]
@majormoses fwiw i'm inclined to agree with @eheydrick here. despite the lack of active development on ruby-mysql, it still works and is unequivocally the simplest solution. the effort to convert to mysql2 is minimal, so it'd be very straightforward to revisit down the road if it were necessary.

**Comment by majormoses** [comment]
Sounds good closing. If we experience pain in the future we can evaluate this then.

**Pull request closed without merging**